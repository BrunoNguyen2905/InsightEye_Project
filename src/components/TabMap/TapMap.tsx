import * as React from "react";
import Tab, { TabProps } from "@material-ui/core/Tab";
import IconButton from "@material-ui/core/IconButton";
import CloseIcon from "@material-ui/icons/Close";
import { withStyles, createStyles } from "@material-ui/core/styles";
import IStyleProps from "../../styles/utils";

const styles = createStyles({
  root: {
    maxWidth: 264,
    position: "relative"
    // paddingTop: 5,
    // paddingBottom: 5
  },
  icon: {
    position: "absolute",
    right: 0,
    top: 0,
    zIndex: 1,
    width: 30,
    height: 30,

    "& svg": {
      fontSize: 15
    }
  }
});

interface IProps extends TabProps {
  canClose?: boolean;
  onClose?: () => void;
}

const TabMap = ({
  classes,
  canClose = false,
  onClose,
  ...props
}: IProps & IStyleProps) => {
  return canClose ? (
    <div className={classes.root}>
      <IconButton className={classes.icon} onClick={onClose}>
        <CloseIcon />
      </IconButton>
      <Tab {...props} />
    </div>
  ) : (
    <Tab {...props} />
  );
};

export default withStyles(styles)(TabMap);
