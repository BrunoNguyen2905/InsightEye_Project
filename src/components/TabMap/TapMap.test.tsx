import * as React from "react";
import { shallow, mount } from "enzyme";
import { spy } from "sinon";
import * as renderer from "react-test-renderer";
import TabMap from "./TapMap";

describe("<TapMap />", () => {
  it("Should render normal mui tab", () => {
    shallow(<TabMap />);
    const tree = renderer.create(<TabMap />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render mui tab with close button", () => {
    shallow(<TabMap canClose={true} />);
    const tree = renderer.create(<TabMap canClose={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should callback onClose when click on icon close", () => {
    const mockOnClick = spy();
    const wrap = mount(<TabMap canClose={true} onClose={mockOnClick} />);
    const buttonIcon = wrap.find("button").first();
    buttonIcon.simulate("click");
    expect(mockOnClick.calledOnce).toEqual(true);
  });
});
