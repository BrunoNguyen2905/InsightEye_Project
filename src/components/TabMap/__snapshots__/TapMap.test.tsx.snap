// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TapMap /> Should render mui tab with close button 1`] = `
<div
  className="TabMap-root-1"
>
  <button
    className="MuiButtonBase-root-15 MuiIconButton-root-25 TabMap-icon-2"
    disabled={false}
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiIconButton-label-30"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-31"
        color={undefined}
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-18"
    />
  </button>
  <button
    aria-selected={undefined}
    className="MuiButtonBase-root-15 MuiTab-root-3 MuiTab-textColorInherit-5"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="tab"
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiTab-wrapper-11"
    />
    <span
      className="MuiTouchRipple-root-18"
    />
  </button>
</div>
`;

exports[`<TapMap /> Should render normal mui tab 1`] = `
<button
  aria-selected={undefined}
  className="MuiButtonBase-root-15 MuiTab-root-3 MuiTab-textColorInherit-5"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="tab"
  tabIndex="0"
  type="button"
>
  <span
    className="MuiTab-wrapper-11"
  />
  <span
    className="MuiTouchRipple-root-18"
  />
</button>
`;
